service/kvmconfig/v6/key/key.go

```diff
@@ -140,6 +143,41 @@ func MemoryQuantity(n v1alpha1.KVMConfigSpecKVMNode) (resource.Quantity, error)
 	return q, nil
 }
 
+// MemoryQuantity returns a resource.Quantity that represents the memory to be used by the nodes.
+// It adds the memory from the node definition parameter to the additional memory calculated on the node role
+func MemoryQuantityWorker(n v1alpha1.KVMConfigSpecKVMNode) (resource.Quantity, error) {
+	mQuantity, err := resource.ParseQuantity(n.Memory)
+	if err != nil {
+		return resource.Quantity{}, microerror.Maskf(err, "calculating memory overhead multiplier")
+	}
+
+	// base worker memory calculated in MB
+	q, err := resource.ParseQuantity(fmt.Sprintf("%dM", mQuantity.ScaledValue(resource.Giga)*1024))
+	if err != nil {
+		return resource.Quantity{}, microerror.Maskf(err, "creating Memory quantity from node definition")
+	}
```

Comment:
I don't understand why this can't just be following?

```go
	m, err := resource.ParseQuantity(n.Memory)
	if err != nil {
		return resource.Quantity{}, microerror.Maskf(err, "calculating memory overhead multiplier")
	}
```

Why this line is needed `q, err := resource.ParseQuantity(fmt.Sprintf("%dM", mQuantity.ScaledValue(resource.Giga)*1024))`?

Reply:
because 8GB doesn't equal 8*1024

Reply:
But that's the value specified by the user. `8G == 8*1000`, `8Gi == 8*1024`.

Reply:
the difference is important here

Reply:
But this means we deal with `G` different way than kubernetes does.

Reply:
we set the limits in `M` not `G` the limits just dont round up nicely with G

Reply:
Whatever, then kvm-operator deal with `M` different way than kubernetes does.
Say at some point kvm-operator starts supporting `Mi` what we do then?

It isn't all that important, because we probably won't support anything else than `M`, but still confusing.

OK. I gave my LGTM. This is tested. I understand what's going on. Important thing is user doesn't know how it's computed and we can change it any time.